templates/create_cloud_conf.sh

```diff
@@ -0,0 +1,110 @@
+else
+  echo "Error: No cloud specified"
+  echo "You must specify which cloud you want to use."
+  echo ""
+  return 1
+fi
+
+CAPO_YQ_TYPE=$(file "$(which yq)")
+if [[ ${CAPO_YQ_TYPE} == *"Python script"* ]]; then
+  echo "Wrong version of 'yq' installed, please install the one from https://github.com/mikefarah/yq"
+  echo ""
+  return 1
+fi
+
+CAPO_CLOUDS_PATH=${CAPO_CLOUDS_PATH:-""}
+CAPO_OPENSTACK_CLOUD_YAML_CONTENT=$(cat "${CAPO_CLOUDS_PATH}")
+
+# Just blindly parse the cloud.yaml here, overwriting old vars.
+CAPO_AUTH_URL=$(echo "$CAPO_OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CAPO_CLOUD}.auth.auth_url)
+CAPO_USERNAME=$(echo "$CAPO_OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CAPO_CLOUD}.auth.username)
+CAPO_PASSWORD=$(echo "$CAPO_OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CAPO_CLOUD}.auth.password)
+CAPO_REGION=$(echo "$CAPO_OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CAPO_CLOUD}.region_name)
+CAPO_PROJECT_ID=$(echo "$CAPO_OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CAPO_CLOUD}.auth.project_id)
+CAPO_PROJECT_NAME=$(echo "$CAPO_OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CAPO_CLOUD}.auth.project_name)
+CAPO_DOMAIN_NAME=$(echo "$CAPO_OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CAPO_CLOUD}.auth.user_domain_name)
+if [[ "$CAPO_DOMAIN_NAME" = "null" ]]; then
+  CAPO_DOMAIN_NAME=$(echo "$CAPO_OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CAPO_CLOUD}.auth.domain_name)
+fi
+CAPO_DOMAIN_ID=$(echo "$CAPO_OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CAPO_CLOUD}.auth.user_domain_id)
+if [[ "$CAPO_DOMAIN_ID" = "null" ]]; then
+  CAPO_DOMAIN_ID=$(echo "$CAPO_OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CAPO_CLOUD}.auth.domain_id)
+fi
+
+# Build cloud.conf
+# Basic cloud.conf, no LB configuration as that data is not known yet.
+CAPO_CLOUD_PROVIDER_CONF_TMP=$(mktemp /tmp/cloud.confXXX)
+cat >> ${CAPO_CLOUD_PROVIDER_CONF_TMP} << EOF
+[Global]
+auth-url=${CAPO_AUTH_URL}
+username="${CAPO_USERNAME}"
+password="${CAPO_PASSWORD}"
+EOF
+
+if [[ "$CAPO_PROJECT_ID" != "" && "$CAPO_PROJECT_ID" != "null" ]]; then
+  echo "tenant-id=\"${CAPO_PROJECT_ID}\"" >> ${CAPO_CLOUD_PROVIDER_CONF_TMP}
+fi
+if [[ "$CAPO_PROJECT_NAME" != "" && "$CAPO_PROJECT_NAME" != "null" ]]; then
+  echo "tenant-name=\"${CAPO_PROJECT_NAME}\"" >> ${CAPO_CLOUD_PROVIDER_CONF_TMP}
+fi
+if [[ "$CAPO_DOMAIN_NAME" != "" && "$CAPO_DOMAIN_NAME" != "null" ]]; then
+  echo "domain-name=\"${CAPO_DOMAIN_NAME}\"" >> ${CAPO_CLOUD_PROVIDER_CONF_TMP}
+fi
+if [[ "$CAPO_DOMAIN_ID" != "" && "$CAPO_DOMAIN_ID" != "null" ]]; then
+  echo "domain-id=\"${CAPO_DOMAIN_ID}\"" >> ${CAPO_CLOUD_PROVIDER_CONF_TMP}
+fi
+
+if [[ "$CAPO_CACERT_ORIGINAL" != "" && "$CAPO_CACERT_ORIGINAL" != "null" ]]; then
+  echo "ca-file=\"${CAPO_CACERT_ORIGINAL}\"" >> ${CAPO_CLOUD_PROVIDER_CONF_TMP}
+fi
+if [[ "$CAPO_REGION" != "" && "$CAPO_REGION" != "null" ]]; then
+  echo "region=\"${CAPO_REGION}\"" >> ${CAPO_CLOUD_PROVIDER_CONF_TMP}
+fi
+
+cat ${CAPO_CLOUD_PROVIDER_CONF_TMP} > /tmp/cloud.conf
```

Comment:
maybe you want to make `/tmp/cloud.conf` a configurable option..

Reply:
Should I update env.rc to add create `/tmp/cloud.conf` option?

Reply:
yes, at least, accept a param instead of hard code /tmp/cloud.conf

Reply:
updated like https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/595/files#diff-8058b1c0be9bf0e46ec0001c6e7c1308R24